### PR TITLE
Add the ability to delete message blocks

### PR DESCRIPTION
**Pull request by EianWright**
This PR contains the changes needed to delete code blocks. We'll figure out the details of the UI for it later on.

### CODE REVIEW

File: webview-ui/src/components/ChatView.tsx [hunk 1]
```diff
@@ -75,14 +75,43 @@ export const ChatView = ({
     );
   };
 
+  const deleteMessageBlock = (
```
Comment (smophat7) [inline webview-ui/src/components/ChatView.tsx:78]: If `deleteMessageBlock` is only used in a `<Message>` component, should it just be defined in `Message.tsx`? This definitely might be a design pattern or React thing I'm unfamiliar with, but I'm just curious.

File: webview-ui/src/components/MessageBlock.tsx [hunk 2]
```diff
@@ -108,3 +128,18 @@ function getThemedSyntaxStyle(): {
...
+
+const DeleteMessageBlockButton = ({
+  deleteMessageBlock,
+}: {
+  deleteMessageBlock: () => void;
+}) => (
+  <VSCodeButton
+    appearance="icon"
+    aria-label="Remove code block"
+    title="Remove code block"
```
Comment (smophat7) [inline webview-ui/src/components/MessageBlock.tsx:140]: The label/title need to be accurate for both text and code blocks

**Review by smophat7 — CHANGES_REQUESTED** [review]
Minor fix. Looks great otherwise!
Styling the trash button on text blocks (location and such) we'll do later, I assume?

File: webview-ui/src/components/ChatView.tsx (hunk 1 above)
Comment (EianWright) [reply webview-ui/src/components/ChatView.tsx:78]: This needs to know where the message block is located and the `<Message>` component didn't have that being passed as props. I would rather just pass it the function and it not have any knowledge of where it was located, than pass it where it is located and have it use that. I don't know if it's the best way to do it. It's just the way that I came up with.
The other solution I thought of was giving every message block a unique id and setting up an endpoint on the extension side which you would send the id and the extension side would handle the logic. This seemed like a much less intrusive solution.

File: webview-ui/src/components/MessageBlock.tsx (hunk 2 above)
Comment (EianWright) [reply webview-ui/src/components/MessageBlock.tsx:140]: Good catch. Thanks!

File: webview-ui/src/components/ChatView.tsx (hunk 1 above)
Comment (smophat7) [reply webview-ui/src/components/ChatView.tsx:78]: Ah yep that makes sense!

**Review by smophat7 — APPROVED** [review]
👍👍

**Pull request merged**